server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1532,6 +1532,22 @@ public CanMatchShardResponse canMatch(ShardSearchRequest request) throws IOExcep
 
     private CanMatchShardResponse canMatch(ShardSearchRequest request, boolean checkRefreshPending) throws IOException {
         assert request.searchType() == SearchType.QUERY_THEN_FETCH : "unexpected search type: " + request.searchType();
+        if (request.source() != null && (request.source().query() instanceof MatchNoneQueryBuilder) == false) {
+            final SearchExecutionContext searchExecutionContext = new SearchExecutionContext(
+                indicesService.indexServiceSafe(request.shardId().getIndex())
+                    .newSearchExecutionContext(
+                        request.shardId().getId(),
+                        request.shardRequestIndex(),
+                        null,
```

Comment:
Using `null` here is okish but we should update the javadoc in [SearchExecutionContext](https://github.com/elastic/elasticsearch/blob/29ecff1802ca31fcb210bd54164723eeab94b646/server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java#L685)?